src/drivers/stm32/adc/adc.cpp

```diff
@@ -467,44 +442,39 @@ ADC::update_system_power(hrt_abstime now)
 
 	/* The valid signals (HW dependent) are associated with each brick */
 #if !defined(BOARD_NUMBER_DIGITAL_BRICKS)
+	irqstate_t flags = px4_enter_critical_section();
```

Comment:
Why?

Reply:
A guess while waiting for a response above. https://github.com/PX4/Firmware/pull/12638#discussion_r310955760